plugin/src/main/java/io/jenkins/plugins/coverage/CoverageProcessor.java

```diff
@@ -227,7 +228,8 @@ private CoverageAction convertResultToAction(final CoverageResult coverageReport
             Ratio buildRatio = coverageReport.getCoverage(coverageElement);
 
             if (buildRatio != null) {
-                float diff = buildRatio.getPercentageFloat() - referenceRatio.getPercentageFloat();
+                //Converting float point to Big decimal and converting back fixes the float precision point problem
+                float diff = BigDecimal.valueOf((buildRatio.getPercentageFloat() - referenceRatio.getPercentageFloat())).floatValue();
```

Comment:
Can you wrap that code into a method? Then you can write a small unit test that verifies the new behavior.

Reply:
Added tests. Please re-review